### PR TITLE
Added ibmcloud IPI conf chain and step for CI

### PR DESCRIPTION
**Pull request by cjschaef**
Added IBM Cloud support for IPI conf chain and step as part of CI
integration.

### CODE REVIEW

**Comment by cjschaef** [comment]
I need to change from `PassThrough` credential mode to `Manual` support and clean up a few of the other config options for latest code changes....

**Review by droslean — APPROVED** [review]

**Comment by droslean** [comment]
@cjschaef Is there anything else to be done in this PR?

**Comment by cjschaef** [comment]
@droslean I believe it sounds like we may need another step added for manual credential mode configuration in the CI.
While we enable the manual mode support using the existing step, it sounds like we need to perform some additional setup for the CI prior to running e2e tests.

I will be attempting to look into that support, as our development team works out the process, and likely create a new step to perform any installation and setup to complete this requirement, then add that new step into this proposed chain.

File: ci-operator/step-registry/ipi/conf/ibmcloud/ipi-conf-ibmcloud-chain.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+chain:
+  as: ipi-conf-ibmcloud
+  steps:
+  - ref: ipi-conf
+  - ref: ipi-conf-ibmcloud
+  - ref: ipi-conf-manual-creds
```
Comment (droslean) [inline ci-operator/step-registry/ipi/conf/ibmcloud/ipi-conf-ibmcloud-chain.yaml:6]: That step only adds `credentialsMode: Manual` to the `install-config.yaml`. Why not just hardcode it here?

Comment (cjschaef) [reply ci-operator/step-registry/ipi/conf/ibmcloud/ipi-conf-ibmcloud-chain.yaml:6]: Yes, I expect we'll add a new step after this to perform any configuration we need for supporting the CI, similar to what AWS does, as a new step to be added once those details are worked out.
https://github.com/openshift/release/blob/7e61829f682e1574513f78c3e94537836d824ab3/ci-operator/step-registry/ipi/aws/pre/manual-oidc-sts/ipi-aws-pre-manual-oidc-sts-chain.yaml#L6

**Comment by cjschaef** [comment]
We are still missing the actual credentials in Vault yet, while waiting for access to add them.

**Comment by cjschaef** [comment]
Closing in favor of compiling with other PR's
- https://github.com/openshift/release/pull/19101
- https://github.com/openshift/release/pull/19100
- https://github.com/openshift/release/pull/20179

**Pull request closed without merging**